logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueReader.java

```diff
@@ -223,6 +244,28 @@ byte[] pollEntryBytes(long timeout) throws IOException, InterruptedException {
         return event;
     }
 
+    /**
+     * Acknowledge last read event, must match every {@code #pollEntry} call.
```

Comment:
I think that's a good idea, the `DeadLetterQueueReader` should collect all the totally consumed segments, and when this `markForDelete` is invoked it wipes out all the completed segments it has collected up to that point.
In particular, the method `pollEntryBytes` when [it store the last segment in the field](https://github.com/elastic/logstash/pull/14188/files#diff-22e56a5df12549ecbf9622646f8aa68040a9d95cda37f5376f8d2f1dbe790358R229-R232) the `lastConsumedReader` it should instead save in a list.
I think that at that point we need a way to impose to the client to invoke `markForDelete` so that the deletion could happen.

Reply:
Maybe we could use the `removeSegmentsBefore(segmentPath);` method?

Reply:
Yes, good idea

Reply:
Implemented with commit 460509adb4d25bb7649a59b2133d01ae7b6b91f3